File: GraduateStudyWebApp/ExamLibSpa_UT/Pages/TS4GP/PubFun/stuLoginPara.ts
import { Dictionary } from "./tzDictionary.js";

export class stuLoginPara {
    
    /// <summary>
    /// 每页记录数
    /// </summary>
    public QxPrjId: string = "";
    /// <summary>
    /// 页序号
    /// </summary>
    public EffectiveDate: string = "";
       
    /// <summary>
    /// 条件串
    /// </summary>
    public LoginName: string = "";
    /// <summary>
    /// 排序表达式
    /// </summary>
    public UserPassword: string = "";

    ///// <summary>
    ///// 排除的检查字符串列表
    ///// </summary>
    //public QxPrjId: Array<string> = new Array<string>();

    ///// <summary>
    ///// 是否检查SQL攻击
    ///// </summary>
    //public IsCheckSQLAttack: boolean = false;

    public static GetMapParam(objLoginPara: stuLoginPara): Dictionary {
        //List < string > arrLst = new List<string>(objLoginPara.QxPrjId);
        var mapParam: Dictionary = new Dictionary();
        mapParam.add("strLoginName", objLoginPara.LoginName);
        mapParam.add("strUserPassword", objLoginPara.UserPassword);
        mapParam.add("intEffectiveDate", objLoginPara.EffectiveDate);
        mapParam.add("intQxPrjId", objLoginPara.QxPrjId);

        return mapParam;
     
    }
}

File: TzPubFunction/com/taishsoft/common/stuLoginPara.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Web;

namespace com.taishsoft.common
{
    /// <summary>
    /// 根据记录序号范围获取对象列表的参数对象-Static
    /// </summary>
    public static class stuLoginPara_Static
    {
        /// <summary>
        /// 根据记录范围获取记录的参数对象获取相关的字典参数
        /// </summary>
        /// <param name="objLoginPara">根据范围获取记录的参数对象</param>
        /// <returns>获取的相关字典参数</returns>
        public static Dictionary<string, string> GetDictParam(this stuLoginPara objLoginPara)
        {
            Dictionary<string, string> dictParam = new Dictionary<string, string>()
            {
                ["strLoginName"] = objLoginPara.LoginName,
                ["strUserPassword"] = "已经隐藏",
                ["strQxPrjId"] = objLoginPara.QxPrjId,
                ["strEffectiveDate"] = objLoginPara.EffectiveDate
            };
            return dictParam;
        }
    }
    /// <summary>
    /// 根据记录序号范围获取对象列表的参数对象
    /// </summary>
    [Serializable]
    [DataContractAttribute]
    public class stuLoginPara
    {     

        /// <summary>
        /// 有效日期
        /// </summary>
        [DataMember]//非常重要
        public string EffectiveDate { get; set; }
        /// <summary>
        /// 当前选择的工程Id
        /// </summary>
        [DataMember]//非常重要
        public string QxPrjId { get; set; }

        /// <summary>
        /// 条件串
        /// </summary>
        [DataMember]//非常重要
        public string LoginName { get; set; }
        /// <summary>
        /// 排序表达式
        /// </summary>
        [DataMember]//非常重要
        public string UserPassword { get; set; }


    }
}